src/main/java/org/openrewrite/java/migrate/DetectAWTGetPeerMethod.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.java.migrate;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.ChangeMethodName;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+@AllArgsConstructor(access = AccessLevel.PACKAGE) // For tests
+class DetectAWTGetPeerMethod extends Recipe {
+    String methodPatternGetPeer;
+    String methodUpdateIsDisplayable;
+    String className;
+    String methodUpdateIsLightweight;
+
+    @Override
+    public String getDisplayName() {
+        return "Replace `getPeer()` method";
+    }
+
+    @Override
+    public String getDescription() {
+        return " All methods that refer to types defined in the java.awt.peer package are removed in Java 11. "
+               + "This recipe replaces the use of getPeer() method in the java.awt.Component, java.awt.Font, and java.awt.MenuComponent classes and direct known subclasse."
+               + "The occurrence of  `(component.getPeer() != null) { .. }` is replaced with `(component.isDisplayable())` "
+               + "and the occurrence of `(component.getPeer() instanceof LightweightPeer)` "
+               + "is replaced with `(component.isLightweight())`.";
+    }
+
+    @JsonCreator
+    public DetectAWTGetPeerMethod() {
+        this.methodPatternGetPeer = "java.awt.* getPeer()";
+        this.methodUpdateIsDisplayable = "java.awt.* isDisplayable()";
+        this.className = "java.awt.peer.LightweightPeer";
+        this.methodUpdateIsLightweight = "java.awt.*  isLightweight()";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesMethod<>(methodPatternGetPeer), new JavaVisitor<ExecutionContext>() {
+            @Override
+            public <T extends J> J.ControlParentheses<T> visitControlParentheses(J.ControlParentheses<T> cp, ExecutionContext ctx) {
```

Comment:
By overriding just `visitControlParentheses` we might miss other cases, as pointed out with an addition to the tests below. Did you already try separate overrides of `visitInstanceOf` & `visitBinary` instead of `visitControlParentheses`?